new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java

```diff
@@ -222,8 +206,288 @@ public void testServerPodsRestartByChangingResource() {
     assertEquals(requests.get("cpu").getNumber().compareTo(cpuRequest), 0,
         String.format("server pod compute resources requests was not updated correctly, set cpu request to %s, got %s",
             cpuRequest, requests.get("cpu").getNumber()));
+
+    // verify the server pods are rolling restarted and back to ready state
+    logger.info("Verifying rolling restart occurred for domain {0} in namespace {1}",
+        domainUid, domainNamespace);
+    assertTrue(assertDoesNotThrow(
+        () -> verifyRollingRestartOccurred(podsWithTimeStamps, 1, domainNamespace),
+        "More than one pod was restarted at same time"),
+        String.format("Rolling restart failed for domain %s in namespace %s", domainUid, domainNamespace));
+
+  }
+
+  /**
+   * Modify the domain scope property on the domain resource.
+   * Verify all pods are restarted and back to ready state.
+   * The resource tested: includeServerOutInPodLog: true --> includeServerOutInPodLog: false.
+   */
+  @Test
+  @DisplayName("Verify server pods are restarted by changing IncludeServerOutInPodLog")
+  @Slow
+  public void testServerPodsRestartByChangingIncludeServerOutInPodLog() {
+    // get the original domain resource before update
+    Domain domain1 = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace),
+        String.format("getDomainCustomResource failed with ApiException when tried to get domain %s in namespace %s",
+            domainUid, domainNamespace));
+    assertNotNull(domain1, domain1 + " is null");
+    assertNotNull(domain1.getSpec(), domain1 + "/spec is null");
+
+    // get the map with server pods and their original creation timestamps
+    podsWithTimeStamps = getPodsWithTimeStamps();
+
+    //print out the original IncludeServerOutInPodLog
+    Boolean includeServerOutInPodLog = domain1.getSpec().getIncludeServerOutInPodLog();
+    logger.info("Original IncludeServerOutInPodLog is: {0}", includeServerOutInPodLog);
+
+    //change includeServerOutInPodLog: true --> includeServerOutInPodLog: false
+    StringBuffer patchStr = null;
+    patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/includeServerOutInPodLog\",")
+        .append("\"value\": ")
+        .append(false)
+        .append("}]");
+    logger.info("PatchStr for includeServerOutInPodLog: {0}", patchStr.toString());
+
+    boolean cmPatched = patchDomainResource(domainUid, domainNamespace, patchStr);
+    assertTrue(cmPatched, "patchDomainCustomResource(IncludeServerOutInPodLog) failed");
+
+    domain1 = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace),
+        String.format("getDomainCustomResource failed with ApiException when tried to get domain %s in namespace %s",
+            domainUid, domainNamespace));
+    assertNotNull(domain1, domain1 + " is null");
```

Comment:
If the domain1 is Null, it will throw a NPE. We can change the statement to
assertNotNull(domain1, "Got Null Domain Resource after patching ");

Reply:
Done